crates/rpc/src/api.rs

```diff
@@ -0,0 +1,420 @@
+
+    #[method(name = "estimateFee")]
+    async fn estimate_fee(
+        &self,
+        request: BroadcastedTransaction,
+        block_id: BlockId,
+    ) -> Result<Vec<FeeEstimate>, BeerusRpcError>;
+
+    #[method(name = "estimateMessageFee")]
+    async fn estimate_message_fee(&self, message: MsgFromL1, block_id: BlockId) -> Result<FeeEstimate, BeerusRpcError>;
+
+    #[method(name = "blockNumber")]
+    async fn block_number(&self) -> Result<u64, BeerusRpcError>;
+
+    #[method(name = "blockHashAndNumber")]
+    async fn block_hash_and_number(&self) -> Result<BlockHashAndNumber, BeerusRpcError>;
+
+    #[method(name = "chainId")]
+    async fn chain_id(&self) -> Result<FieldElement, BeerusRpcError>;
+
+    #[method(name = "pendingTransactions")]
+    async fn pending_transactions(&self) -> Result<Vec<Transaction>, BeerusRpcError>;
+
+    #[method(name = "syncing")]
+    async fn syncing(&self) -> Result<SyncStatusType, BeerusRpcError>;
+
+    #[method(name = "getEvents")]
+    async fn get_events(
+        &self,
+        filter: EventFilter,
+        continuation_token: Option<String>,
+        chunk_size: u64,
+    ) -> Result<EventsPage, BeerusRpcError>;
+
+    #[method(name = "getNonce")]
+    async fn get_nonce(
+        &self,
+        block_id: BlockId,
+        contract_address: FieldElement,
+    ) -> Result<FieldElement, BeerusRpcError>;
+
+    #[method(name = "addInvokeTransaction")]
+    async fn add_invoke_transaction(
+        &self,
+        invoke_transaction: BroadcastedInvokeTransaction,
+    ) -> Result<InvokeTransactionResult, BeerusRpcError>;
+
+    #[method(name = "addDeclareTransaction")]
+    async fn add_declare_transaction(
+        &self,
+        declare_transaction: BroadcastedDeclareTransaction,
+    ) -> Result<DeclareTransactionResult, BeerusRpcError>;
+
+    #[method(name = "addDeployAccountTransaction")]
+    async fn add_deploy_account_transaction(
+        &self,
+        deploy_account_transaction: BroadcastedDeployAccountTransaction,
+    ) -> Result<DeployAccountTransactionResult, BeerusRpcError>;
+
+    #[method(name = "estimateFeeSingle")]
+    async fn estimate_fee_single(
+        &self,
+        request: BroadcastedTransaction,
+        block_id: BlockId,
+    ) -> Result<FeeEstimate, BeerusRpcError>;
+
+    // ------------------- Extended Starknet Provider Endpoints -------------------
+    //
+    #[method(name = "getContractStorageProof")]
+    async fn get_contract_storage_proof(
+        &self,
+        block_id: BlockId,
+        contract_address: FieldElement,
+        keys: Vec<FieldElement>,
+    ) -> Result<StorageProof, BeerusRpcError>;
+
+    #[method(name = "provenStateRoot")]
+    async fn proven_state_root(&self) -> Result<FieldElement, BeerusRpcError>;
+
+    #[method(name = "provenBlockNumber")]
+    async fn proven_block_number(&self) -> Result<u64, BeerusRpcError>;
```

Comment:
It looks like the difference between `block_number()` and `proven_block_number()` is Pool seconds and additional node call, i mean maybe it's not necessary endpoint?

Reply:
yup will delete